Assets/Scripts/Data/Pickup Pup/Currency/CurrencyData.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * Author: Grace Barrett-Snyder, Isaiah Mann
+ * Description: Holds the data for a specific form of currency (ex: coins)
+ */
+
+using System.IO;
+using UnityEngine;
+
+[System.Serializable]
+public class CurrencyData
+{
+    #region Instance Accessors
+
+    public int Amount
+    {
+        get
+        {
+            return amount;
+        }
+    }
+
+    public CurrencyType Type
+    {
+        get
+        {
+            return type;
+        }
+    }
+
+    public virtual Sprite Icon
+    {
+        get
+        {
+            return Resources.Load<Sprite>(Path.Combine("Sprites", "Default"));
```

Comment:
If it doesn't need a different superclass, set its superclass to `ResourceLoader` and use the constants instead of hardcoded strings. Otherwise, add those constants to whatever superclass it needs instead